sasmodels/kernelcuda.py

```diff
@@ -137,6 +138,14 @@ def environment():
             raise RuntimeError("SAS_OPENCL=None in environment")
     return ENV
 
+def free_context():
+    global ENV
+    if ENV is not None:
+        ENV.release()
+        ENV = None
+
+atexit.register(free_context)
+
```

Comment:
pep8 recommends two blank lines before and after a top level function definition.

On a more important note, I couldn't see why the atexit code was necessary.  Shouldn't the ENV's destructor call release when the module is unloaded at shutdown?  I know that some exceptions will stop the destructor from being called, but I also thought that most of those would stop atexit.  I'm not quite as familiar with the atexit module, though, so there may be an advantage that I'm missing here.

Reply:
Yes, you are correct.  I presumably introduced it before adding the __del__ to the environment.

Reply:
Upon testing, it appears that pytest leaves ENV hanging around, so I reintroduced the atexit.